### PR TITLE
added packages/bytearray/bytearray.1.0.0

### PR DESCRIPTION
**Pull request by UnixJunkie**

### CODE REVIEW

**Comment by UnixJunkie** [comment]
Probably do not merge.
I'll have to make sure this compiles on most OCaml versions.
I want to see the CI tests matrix to know where and what to do.
This code was extracted out of parmap.

File: packages/bytearray/bytearray.0.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+authors: "Jérôme Vouillon <Jerome.Vouillon@pps.univ-paris-diderot.fr>"
+maintainer: "unixjunkie@sdf.org"
+homepage: "https://github.com/UnixJunkie/bytearray"
+dev-repo: "git+https://github.com/UnixJunkie/bytearray.git"
+bug-reports: "https://github.com/UnixJunkie/bytearray/issues"
+build: [
+  ["dune" "build" "-p" name]
+]
+depends: [
+  "ocaml"
```
Comment (kit-ty-kate) [inline packages/bytearray/bytearray.0.0.1/opam]:
```suggestion
  "ocaml"
  "dune"
```
This seems to require dune

File: packages/bytearray/bytearray.0.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,18 @@
+opam-version: "2.0"
+authors: "Jérôme Vouillon <Jerome.Vouillon@pps.univ-paris-diderot.fr>"
+maintainer: "unixjunkie@sdf.org"
+homepage: "https://github.com/UnixJunkie/bytearray"
+dev-repo: "git+https://github.com/UnixJunkie/bytearray.git"
+bug-reports: "https://github.com/UnixJunkie/bytearray/issues"
+build: [
+  ["dune" "build" "-p" name]
```
Comment (kit-ty-kate) [inline packages/bytearray/bytearray.0.0.1/opam]:
```suggestion
  ["dune" "build" "-p" name "-j" jobs]
```
Allow users to set the number of jobs allocated to build this is always good

**Comment by UnixJunkie** [comment]
I have to detect the OCaml version, use a preprocessor and not use Unix.map_file before 4.06.0.

**Comment by UnixJunkie** [comment]
Only the mac os CI has stalled. I will update the tag to v1.0.0.0 then.

**Comment by UnixJunkie** [comment]
All CI tests are green.

**Comment by mseri** [comment]
Thanks

**Pull request merged**